src/AddressDriver.sol

```diff
@@ -94,6 +94,27 @@ contract AddressDriver is Upgradeable, ERC2771Context {
     /// Positive to add funds to the drips balance, negative to remove them.
     /// @param newReceivers The list of the drips receivers of the sender to be set.
     /// Must be sorted by the receivers' addresses, deduplicated and without 0 amtPerSecs.
+    /// @param maxEndTip1 The first tip to calculate the maximum end time of drips
+    /// after the balance and the receivers list are updated.
+    /// Tips have no effect on the results of calling this function, but they can reduce
+    /// the range of timestamps when searching for the maximum end time, which saves gas.
+    /// Tips lower than the current timestamp are ignored,
+    /// and passing a zero is an easy way to not provide a tip.
+    /// It's acceptable to provide zero, one or two tips. The order of tips doesn't matter.
+    /// Tips are the most effective when one of them is lower than or equal to the maximum end time,
+    /// and the other one is strictly larger than the maximum end time,
+    /// the smaller the difference between them, the higher gas savings.
+    /// The savings are the highest possible when one of the tips
+    /// is equal to the maximum end time and the other one is larger by 1.
+    /// It's worth noting that the exact timestamp of the block in which this function is executed
+    /// may affect correctness of the tips, especially if they're precise.
+    /// Tips are used to divide the binary searched range of future timestamps into smaller parts,
```

Comment:
```suggestion
    /// Tips are used to find the maxEnd (running out of funds) time with a binary search by dividing the search range of future timestamps into smaller parts,
```

Maybe?

Reply:
I agree, that's clearer, applied something similar.